code/__HELPERS/unsorted.dm

```diff
@@ -781,30 +781,32 @@ Turf and target are seperate in case you want to teleport some distance from a t
 
 	return 1
 
-/proc/do_after(mob/user, delay, numticks = 5, needhand = 1)
-	if(!user || isnull(user))
+/proc/do_after(mob/user, delay, numticks = 5, needhand = 1, atom/target = null)
+	if(!user)
 		return 0
+
 	if(numticks == 0)
 		return 0
 
+	var/atom/Tloc = null
+	if(target)
+		Tloc = target.loc
+
 	var/delayfraction = round(delay/numticks)
-	var/turf/T = user.loc
+	var/atom/T = user.loc

```

Comment:
rename this to Uloc please, to now be in line with your new Tloc.
(since T and Tloc is confusing)